core/src/main/java/fi/iki/elonen/NanoHTTPD.java

```diff
@@ -1196,6 +1196,22 @@ private void decodeHeader(BufferedReader in, Map<String, String> pre, Map<String
                 System.out.println(this + " - decodeHeader: " + inLine);
                 
                 StringTokenizer st = new StringTokenizer(inLine);
+                
+                // check if we are dealing with a request that went through the proxy protocol
+                String proxy = inLine.substring(0, 5).toLowerCase();
```

Comment:
I'd move this to after `hasMoreTokens()` check in case it's empty and reusing the same logic that is used to pull in the method `pre.put("method", st.nextToken());` and check if the "next token" is `proxy`.